src/Umbraco.Core/Services/DataTypeService.cs

```diff
@@ -335,14 +354,12 @@ public IEnumerable<EntityContainer> GetContainers(int[] containerIds)
         /// </summary>
         /// <param name="id">Unique guid Id of the DataType</param>
         /// <returns><see cref="IDataType"/></returns>
-        public IDataType? GetDataType(Guid id)
-        {
-            using ICoreScope scope = ScopeProvider.CreateCoreScope(autoComplete: true);
-            IQuery<IDataType> query = Query<IDataType>().Where(x => x.Key == id);
-            IDataType? dataType = _dataTypeRepository.Get(query).FirstOrDefault();
-            ConvertMissingEditorOfDataTypeToLabel(dataType);
-            return dataType;
-        }
+        public IDataType? GetDataType(Guid id) =>
+            _idKeyMap.Value.GetIdForKey(id, UmbracoObjectTypes.DataType) switch
```

Comment:
We should probably add a little comment to explain why we're using the `IIdKeyMap` here:

```suggestion
        public IDataType? GetDataType(Guid id)
            // Lookup the integer ID, so the data type can be retrieved from the repository cache
            =>_idKeyMap.Value.GetIdForKey(id, UmbracoObjectTypes.DataType) switch
```